### ISSUE TITLE
Conditional false positives? 

### ISSUE THREAD

**Issue by moltar** [veritem/eslint-plugin-vitest#282]
The following code:

```ts
Object.assign({}, {} ?? {})
```

Triggers:

```
vitest/no-conditional-tests
vitest/no-conditional-in-test
```

Is this intended?

I understand that `??` _is_ a conditional, of course.

But it's certainly not a conditional test.

**Comment by veritem** [comment]
provide a full repro, I don't get the context of where you're trying to use this.

**Comment by moltar** [comment]
- repro repo: https://github.com/moltar/eslint-plugin-vitest-repro
- file: https://github.com/moltar/eslint-plugin-vitest-repro/blob/master/tests/t.test.ts
- config: https://github.com/moltar/eslint-plugin-vitest-repro/blob/db1cabd177cf7c9b3d7c5d3cea99b338fc19ef77/.eslintrc.json#L13-L24

Error:

```
> eslint '**/*.ts'


.../eslint-plugin-vitest-repro/tests/t.test.ts
  4:21  error  Avoid using conditionals in a test  vitest/no-conditional-tests
  4:21  error  Remove conditional tests            vitest/no-conditional-in-test

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by veritem** [comment]
Well, our current tests check whether or not there is a presence of conditionals in a test. If you dislike that behavior you can turnoff rules. You can also rewrite your code behavior to look like this

```ts
  import { expect, test } from 'vitest'
 
 Object.assign({}, {} ?? {})

test('a', () => {

  expect(true).toBeTruthy()
})```

**Comment by moltar** [comment]
But isn't there a semantic difference between `conditional-tests` and `conditional-in-test`?

My example can certainly qualify as a `conditional-in-test`.

But it does not make the test itself conditional. The test runs either way.

https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/no-conditional-tests.md

This is the example given:

```ts
describe('my tests', () => {
	if (true) {
		it('is awesome', () => {
			doTheThing()
		})
	}
})
```
